doc/rst/language/spec/variables.rst

```diff
@@ -229,39 +267,97 @@ initialization, not assignment, of that variable.
       5
 
 
-Split initialization does not apply:
-
- * when the variable is a field, config variable, or ``extern`` variable.
- * when an applicable assignment statement setting the variable could not
-   be identified
- * when an applicable assignment statement is in one branch of a
-   conditional but not in the other, unless:
+   *Example (split-init-select-dce.chpl)*
 
-     * the variable is not an ``out`` intent formal, and
-     * the other branch always returns or throws.
+   When the condition for a ``select`` or conditional statement
+   can be evaluated at compile-time, the compiler only considers
+   the path that will be taken:
+    
+   .. code-block:: chapel
+      
+      proc splitInitsBecauseDCE(type T) {
+        var x: int;
+        select T {
+          when int {
+            // split initialization occurs here when T==int
+            x = 1; 
+          }
+          when string {
+            writeln("compiler ignores this block when T==int");
+          }
+        }
+        writeln(x);
+      }
+      proc noSplitInitBecausePathUnknown(arg: int) {
+        var x: int;
+        select arg {
+          when 0 {
+          // no split init, not all paths return, throw, or initialize
+            x = 1; 
+          }
+          when 1 {
+            writeln("this block not ignored by compiler.");
+            writeln("arg value unknown at compile-time.");
+          } 
+        }
+        writeln(x);
+      }
+      proc main() {
+        splitInitsBecauseDCE(int);
+        noSplitInitBecausePathUnknown(0);
+      }
+    
+   .. BLOCK-test-chapeloutput
 
-   This rule prevents split-initialization when the applicable assignment
-   statement is in a conditional that has no ``else`` branch and the
-   ``if`` branch does not return or throw.
+      1
+      1
 
- * when an applicable assignment statement is in a ``try`` or ``try!``
-   block which has ``catch`` clauses that mention the variable
+   *Example (split-init-select-otherwise.chpl)*
 
- * when an applicable assignment statement is in a ``try`` or ``try!``
-   with ``catch`` clauses unless:
+   Remember to consider the path taken by select statements where
```

Comment:
It feels a bit of a corner case that maybe does not need to be remembered all the time?

I think I'd prefer wording that simply describes what happens here. E.g.

For split initialization with a `select` statement, the `otherwise` clause (or absence of an `otherwise` clause) impacts whether or not split initialization is possible within the `when` clauses: